src/x509.cpp

```
@@ -31,17 +34,29 @@ using namespace openssl;
 
 X509Certificate X509Certificate::fromPEM(const std::string &pem)
 {
+    std::string formattedPem = util::reformatPEMCertificate(pem);
     BioObject bio{BioObject::Types::MEM};
-    bio.write(pem);
+    bio.write(formattedPem);
     auto cert = _PEM_read_bio_X509(bio.internal());
     return X509Certificate{std::move(cert)};
 }
 
 X509Certificate X509Certificate::fromPEMFile(const std::string &filename)
 {
-    FileBio bio{filename, FileBio::FileMode::READ, FileBio::FileType::TEXT};
-    auto cert = _PEM_read_bio_X509(bio.internal());
-    return X509Certificate{std::move(cert)};
+    // Change to be reverted to previous solution after OpenSSL parsing
+    // bug for PEM certificates has been fixed.
+    std::ifstream pemFile(filename);
+    std::stringstream buffer;
+    if (pemFile.is_open()) {
+        buffer << pemFile.rdbuf();
+        pemFile.close();
+    } else {
+        throw MoCOCrWException("Couldn't open certificate at: " + filename);
```

Are the MoCOCrWExceptions here an ABI change, i.e. did this throw MoCOCrWExceptions before?

---

hmm, right.. i guess not. It guess std::runtime_error will also do (that actually could have been thrown earlier)

---

@garw What's your opinion on that? Not sure std::runtime_error is a good solution here either. Maybe we should just invoke the OpenSSL functions with empty data and let that throw an error?